BUILD.md

```diff
@@ -221,15 +221,15 @@ Ideally, pull requests targeting performance improvements should include such a
 
 ## Code Coverage
 
-Code coverage of the Tosca project is a work in progress. As the different systems get integrated, more features will be added and documented here.
-Go has native support for coverage, which makes getting this metric for `lfvm` or `geth` driver runs quite simple.
+The Tosca project allows to collect coverage reports for unit tests and CT runs. 
+More about coverage in [the docs folder](docs/coverage.md)
 
 ### CT Driver Code coverage
 
 To run the the CT driver code coverage simply run `make ct-coverage-lfvm` or `make ct-coverage-geth`, these commands will create the folder `Tosca/go/build/coverage/` and a sub folder with the corresponding interpreter's name. In this directory an instrumented version of the driver will be built and then run. The reports of the coverage of this run will also be in this directory, where lastly an HTML version of the report will be added as well. As more interpreter implementations are added to the Tosca project, they should also be added to the makefile.
 
 
-### C++ Code coverage
+### C++ Code Unit Test Coverage
```

Comment:
Long story short. The coverage thing is too complicated not to explain some details. I did not find the right place to write this down for future me or other people. 

The idea behind splitting this documentation is to differentiate between how to run things from some text explaining how does it work internally. I thought it was just too much text to keep it in the same place. 
future changes or improvements in the usage of the feature would lead to `doc/coverage.md` being extended, but the section in `build.md` would need to be rewritten. 

wdyt?